helm_tests/webserver/test_rpc_server.py

```diff
@@ -0,0 +1,923 @@
+            "readOnly": True,
+            "subPath": "airflow.cfg",
+        } in jmespath.search("spec.template.spec.containers[0].volumeMounts", docs[0])
+
+    def test_rpc_server_resources_are_configurable(self):
+        docs = render_chart(
+            values={
+                "rpcServer": {
+                    "resources": {
+                        "limits": {"cpu": "200m", "memory": "128Mi"},
+                        "requests": {"cpu": "300m", "memory": "169Mi"},
+                    }
+                },
+            },
+            show_only=["templates/rpc-server/rpc-server-deployment.yaml"],
+        )
+        assert "128Mi" == jmespath.search("spec.template.spec.containers[0].resources.limits.memory", docs[0])
+        assert "200m" == jmespath.search("spec.template.spec.containers[0].resources.limits.cpu", docs[0])
+
+        assert "169Mi" == jmespath.search(
+            "spec.template.spec.containers[0].resources.requests.memory", docs[0]
+        )
+        assert "300m" == jmespath.search("spec.template.spec.containers[0].resources.requests.cpu", docs[0])
+
+        # initContainer wait-for-airflow-migrations
+        assert "128Mi" == jmespath.search(
+            "spec.template.spec.initContainers[0].resources.limits.memory", docs[0]
+        )
+        assert "200m" == jmespath.search("spec.template.spec.initContainers[0].resources.limits.cpu", docs[0])
+
+        assert "169Mi" == jmespath.search(
+            "spec.template.spec.initContainers[0].resources.requests.memory", docs[0]
+        )
+        assert "300m" == jmespath.search(
+            "spec.template.spec.initContainers[0].resources.requests.cpu", docs[0]
+        )
+
+    def test_rpc_server_security_contexts_are_configurable(self):
+        docs = render_chart(
+            values={
+                "rpcServer": {
+                    "securityContexts": {
+                        "pod": {
+                            "fsGroup": 1000,
+                            "runAsGroup": 1001,
+                            "runAsNonRoot": True,
+                            "runAsUser": 2000,
+                        },
+                        "container": {
+                            "allowPrivilegeEscalation": False,
+                            "readOnlyRootFilesystem": True,
+                        },
+                    }
+                },
+            },
+            show_only=["templates/rpc-server/rpc-server-deployment.yaml"],
+        )
+        assert {"allowPrivilegeEscalation": False, "readOnlyRootFilesystem": True} == jmespath.search(
+            "spec.template.spec.containers[0].securityContext", docs[0]
+        )
+
+        assert {
+            "runAsUser": 2000,
+            "runAsGroup": 1001,
+            "fsGroup": 1000,
+            "runAsNonRoot": True,
+        } == jmespath.search("spec.template.spec.securityContext", docs[0])
+
+    def test_rpc_server_security_context_legacy(self):
```

Comment:
We don't need to support the old style with this new component. If it's too much of a pain with how the helpers are written, maybe we keep it, but lets see if we can whack it.

Reply:
so i figured out that we can disallow it by removing it from the json schema.  no need to mess with the helpers.

now we test that it is forbidden.  but i had to improve the way we invoke the helm command to make that possible, and i split that out into pr https://github.com/apache/airflow/pull/39698